action.yml

```diff
@@ -0,0 +1,40 @@
+name: 'Upload mender artifact'
+description: 'Upload a mender artifact to a hosted Mender backend instance'
+inputs:
+  mender_pat:
+    description: 'personal access token for the hosted Mender backend account to be used'
+    required: true
+  mender_artifact:
+    description: 'path of the artifact to be uploaded, relative to GITHUB_WORKSPACE'
+    required: true
+  mender_uri:
+    description: 'URI for the hosted Mender backend account to be used'
+    required: false
+    default: 'https://hosted.mender.io'
+runs:
+  using: "composite"
+  steps:
+    - id: check_file
+      shell: bash
+      run: |
+        ls ${GITHUB_WORKSPACE}/${{ inputs.mender_artifact }} > /dev/null
+    - id: upload_artifact
+      shell: bash
+      run: |
+        RESPONSE=$(curl -s -X POST ${{ inputs.mender_uri }}/api/management/v1/deployments/artifacts \
```

Comment:
Use `mender-cli artifacts upload` instead

Reply:
Well, it's a topic for a discussion how to proceed with this action here.

If we depend from `mender-cli` tool then we can't create a a more unified [composite](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action) action, because it will depend from provided by us image and limits the action usage. 

Actually it's possible to achieve that with using [docker container](https://docs.github.com/en/actions/creating-actions/creating-a-docker-container-action) action, but it pretty significantly increases action execution time, because it requires to run and initialise another container.

According to the commits log it looks like that @TheYoctoJester already tried that approach and stopped on using composite action instead.

My personal preference is composite action in this case, because it's commonly used API call, which we have basically from day one and unlikely to be changed, and I don't see any need in using special tools for doing the API call. It might be reasonable from maintenance point of view (if the API changes then we adjust `mender-cli` and don't have to fix it here), but I don't think it's the case here.

Reply:
(take into consideration that I am no GH Actions expert),

One goal of the epic is to use our Docker image containing our tools in all possible CIs, so that the interface looks familiar when moving across CIs.

Hence, I vote for using `mender-cli` from our image.  If you are confident that this is a bad idea on GH Actions ecosystem, and that a `composite` action is the best, then I support your decision :+1: